internal/oauthutil/oauth2.go

```diff
@@ -27,6 +27,9 @@ type OAuthContext struct {
 	Endpoint Endpoint
 	// Scope specifies optional requested permissions.
 	Scopes []string
+
+	// CustomURLArgs if any needed by a specific provider that may have a custom implementation.
```

Comment:
I cannot get the meaning of the comment, maybe it can be changed a bit?
```suggestion
	// CustomURLArgs are URL parameters which may be needed by a specific provider that may have a custom OAuth2 implementation.
```